theory_label_generator.py

```diff
@@ -278,14 +289,26 @@ def run_theorem_prover(theorem_prover, ip, ip_format, op, report_metrics):
                     metrics.update(
                         gold_label, engine_label, returned_exception, elapsed_millisecs
                     )
-                instance_json["label"] = engine_label
-                json.dump(instance_json, op)
-                op.write("\n")
```

Comment:
What was the problem with unconditionally writing a newline after every json.dump() call?